### DISCUSSION TITLE
My windows go above sketchybar with yabai

### DISCUSSION THREAD

**Discussion by dlynz** [FelixKratz/SketchyBar#416] [Q&A] [answered]
<img width="1440" alt="image" src="https://github.com/FelixKratz/SketchyBar/assets/100419289/04b9ad83-8677-41c2-a62c-90699705fcc9">
Yabai doesn't seem to be detecting it.

**Comment by FelixKratz** [comment, marked as answer]
Check the `external_bar` property of yabai, it will make sure no windows are put where the bar resides, e.g. when sketchybar and yabai are already running:
```bash
yabai -m config external_bar all:$(sketchybar --query bar | jq .height):0
```

**Comment by fbuetler** [comment]
Any idea on how to solve this for aerospace?

EDIT:

This can be solved in aerospace with the following config

```
[gaps]
outer.bottom = 30
```

While the value should (to avoid having an gap between bar and window) match the height of sketchybar i.e.:

```
sketchybar --bar height=30
```

**Comment by fayimora** [comment]
The Aerospace solution only works for me when using my MacBook's display.

When i'm on my Mac's display, I can keep the gap at 0 and everything is properly aligned.

If I connect an external monitor and close the Mac, everything overlaps.

If I add the gap, everything is fine on external display but the moment I switch back to the Mac's display, I now have the opposite offset problem that looks like this.
![CleanShot 2025-02-14 at 11 14 53@2x](https://github.com/user-attachments/assets/b1c2c1ad-e37d-4a01-a2c0-842cbf41f7d6)


If it helps, i'm on Sequoia 15.3.1

**Reply by kevinmost** [reply]
On https://nikitabobko.github.io/AeroSpace/guide, it indicates that you can set per-monitor gaps:

```
# - Per monitor:  gaps.outer.top = [{ monitor.main = 16 }, { monitor."some-pattern" = 32 }, 24]
```

Something like this worked nicely in my `aerospace.toml` for both the internal notched monitor on my Macbook + my external 4K monitor:

```
[gaps]
    outer.top = [
        # Minimal gap on internal display
        { monitor."^built-in retina display$" = 4 },
        # Larger gap on all other displays
        40,
]
```

**Reply by fayimora** [reply]
Thanks for this @kevinmost. Worked like a charm!
